tests/check/check_test.go

```diff
@@ -33,6 +37,130 @@ func TestCheckMySQL(t *testing.T) {
 	testRunAll(t, "mysql")
 }
 
+func TestCheckLogs(t *testing.T) {
+	cfg := run.MustDefaultConfigWithRandomPorts()
+	cfg.Log.Level = "none"
+	cfg.Datastore.Engine = "memory"
+
+	observerLogger, logs := observer.New(zap.DebugLevel)
+	serverCtx := &run.ServerContext{
+		Logger: &logger.ZapLogger{
+			Logger: zap.New(observerLogger),
+		},
+	}
+
+	cancel := tests.StartServerWithContext(t, cfg, serverCtx)
```

Comment:
I would add a comment here that the reason why we have to start up a full fledged server is because otherwise the logging interceptor (or any other interceptor) will not be hooked up.